src/python/tests/core/bot/fuzzers/afl/afl_launcher_integration_test.py

```diff
@@ -333,126 +281,53 @@ def test_fuzz_merge(self, mock_get_timeout):
-    # *WARNING* Do not lower the fuzz timeout unless you really know what you
-    # are doing. Doing so will cause the test to fail on rare ocassion, which
-    # will break deploys.
-    mock_get_timeout.return_value = get_fuzz_timeout(90.0)
+  def test_fuzz_input_crash(self, mock_get_timeout):
+    """Tests fuzzing (crash in input)."""
+    mock_get_timeout.return_value = get_fuzz_timeout(5.0)
     testcase_path = setup_testcase_and_corpus('empty', 'corpus', fuzz=True)
-
-    output = run_launcher(testcase_path, 'easy_crash_fuzzer')
+    output = run_launcher(testcase_path, 'always_crash_fuzzer')
     self.assertIn(
-        'Command: {0}/afl-fuzz -i/corpus '
-        '-o/afl_output_dir -mnone '
-        '{0}/easy_crash_fuzzer 2147483647'.format(DATA_DIRECTORY), output)
-
-    self.assertIn('ERROR: AddressSanitizer: heap-use-after-free on address',
-                  output)
+        'Command: {0}/afl-fuzz -mnone -o{1}/temp-1337/afl_output_dir '
+        '-i{1}/corpus {0}/always_crash_fuzzer 2147483647'.format(
+            DATA_DIRECTORY, TEMP_DIRECTORY), output)
 
-    # Testcase should've been copied back.
-    self.assertGreaterEqual(os.path.getsize(testcase_path), 3)
-    with open(testcase_path) as f:
-      self.assertEqual(f.read()[:3], 'ABC')
+    self.assertIn(
+        'ERROR: AddressSanitizer: SEGV on unknown address '
+        '0x000000000000', output)
 
-  @no_errors
-  @mock.patch('bot.fuzzers.afl.launcher.get_fuzz_timeout')
-  def test_fuzz_merge(self, mock_get_timeout):
-    """Tests fuzzing with merge."""
-    mock_get_timeout.return_value = get_fuzz_timeout(5.0)
-    testcase_path = setup_testcase_and_corpus(
-        'empty', 'redundant_corpus', fuzz=True)
-    corpus_path = os.environ['FUZZ_CORPUS_DIR']
-    for i in xrange(100):
-      with open(os.path.join(corpus_path, '%04d' % i), 'w') as f:
-        f.write('A' * 256)
+    # Testcase (non-zero) should've been copied back.
+    self.assertNotEqual(os.path.getsize(testcase_path), 0)
 
+  def test_single_testcase_crash(self):
+    """Tests launcher with a crashing testcase."""
+    testcase_path = setup_testcase_and_corpus('crash', 'empty_corpus')
     output = run_launcher(testcase_path, 'test_fuzzer')
     self.assertIn(
-        'Command: {0}/afl-fuzz -d -i/redundant_corpus '
-        '-o/afl_output_dir -mnone '
-        '{0}/test_fuzzer 2147483647'.format(DATA_DIRECTORY), output)
-
-    self.assertIn('Merging corpus.', self.logged_messages)
-    self.assertNotIn('Timed out in merge', ' '.join(self.logged_messages))
-    self.assertIn('Merge completed successfully.', self.logged_messages)
+        'ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000',
+        output)
 
-  @no_errors
-  @mock.patch('bot.fuzzers.afl.launcher.get_fuzz_timeout')
-  def test_libfuzzerize_corpus(self, mock_get_timeout):
-    self._test_libfuzzerize_corpus(mock_get_timeout)
+    # Make sure we didn't fuzz.
+    self.assertNotIn('afl-fuzz', output)
```

Comment:
maybe add an assertEqual on exists and non-zero testcase_path size. we want to ensure that running a testcase does not cause any it to move anywhere ?

Reply:
Done.